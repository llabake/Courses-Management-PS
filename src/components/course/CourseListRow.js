/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const CourseListRow = ({ course, onDelete }) => {
  const deleteCourse = () => (
    onDelete(course.id)
  );
  return (
    <tr>
      <td><a href={`/course/${course.watchHref}`} target="_blank" rel="noopener noreferrer">Watch</a></td>
      <td><Link to={`/course/${course.id}`}>{course.title}</Link></td>
      <td>{course.authorId}</td>
      <td>{course.category}</td>
      <td>{course.length}</td>
      <td
        className="btn btn-xs btn-danger"
        onClick={deleteCourse}
      >
Delete
      </td>
    </tr>
  );
};

CourseListRow.propTypes = {
  course: PropTypes.object.isRequired,
  onDelete: PropTypes.func.isRequired,
};

export default CourseListRow;
