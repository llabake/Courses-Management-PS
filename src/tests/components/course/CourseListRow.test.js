import React from 'react';
import { shallow } from 'enzyme';
import CourseListRow from "../../../components/course/CourseListRow";
import courseMock from "../../mocks/courseMock";

describe('<CourseListRow />', () => {
  const props = {
    onDelete: jest.fn(),
    course: courseMock.createdCourseWithId,
  };
  const wrapper = shallow(<CourseListRow {...props} />);
  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('deletes a course', () => {
    wrapper.find('.btn').simulate('click');
    expect(props.onDelete.mock.calls.length).toEqual(1);
  });
});
