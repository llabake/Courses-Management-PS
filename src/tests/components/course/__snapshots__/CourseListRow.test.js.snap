// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CourseListRow /> renders correctly 1`] = `
<tr>
  <td>
    <a
      href="/course/http://www.pluralsight.com/courses/Learn-Ruby:-Write-Great-Ruby-Code"
      rel="noopener noreferrer"
      target="_blank"
    >
      Watch
    </a>
  </td>
  <td>
    <Link
      replace={false}
      to="/course/Learn-Ruby:-Write-Great-Ruby-Code"
    >
      Learn Ruby: Write Great Ruby Code
    </Link>
  </td>
  <td>
    cory-house
  </td>
  <td>
    Software Practices
  </td>
  <td>
    3:10
  </td>
  <td
    className="btn btn-xs btn-danger"
    onClick={[Function]}
  >
    Delete
  </td>
</tr>
`;
